### ISSUE TITLE
`no-try-expect` is still part of the `jest/recommended`

### ISSUE THREAD

**Issue by alexilyaev** [jest-community/eslint-plugin-jest#678]
To continue https://github.com/jest-community/eslint-plugin-jest/issues/648

@G-Rath Looks like [no-try-expect](https://github.com/jest-community/eslint-plugin-jest/blob/master/src/rules/no-try-expect.ts) is still in the `recommended` list (v24 major was released recently).

I got here because I'm using [eslint-find-rules](https://github.com/sarbbottam/eslint-find-rules) to find deprecated rules in my configs and it failed for `jest/recommended`.

I'm assuming the `recommended` property needs to be removed from the rule file.  
But also thinking if this can be prevented in the future.  
Although if we deprecate a rule and still keep it in the `jest/recommended`, then I'm not sure how.

Any way, here's how I ran `eslint-find-rules`:

```js
// .eslintrc.js
module.exports = {
  plugins: ['jest'],
  extends: ['plugin:jest/recommended'],
};
```

```shell
$ npx eslint-find-rules --deprecated jest-bug.js

deprecated rules

jest/no-try-expect
```

**Comment by G-Rath** [comment]
@alexilyaev this is because we forgot to remove the deprecated rules in the new major.

We can't change what's in `recommended` without a new major, so while we addresses this for `all` when it was reported, we can't do anything in this case without technically breaking semver.

See #663 for related discussion.

**Comment by alexilyaev** [comment]
I understand that.  
But ideally there would be an automated process to check for such a thing, to alleviate the need to not forget to remove rules from `all` and `recommended`.  
Or at least a formal manual step-by-step guide for major releases, noting the necessary things to go through.

**Comment by G-Rath** [comment]
Thanks for understanding.

I'm going to close this for now since we've got #661 already prepared for the next major :)
